src/QSymbolicsBase/basic_ops_homogeneous.jl

```diff
@@ -128,7 +128,7 @@ AB
     terms
     function SMulOperator(terms)
         coeff, cleanterms = prefactorscalings(terms)
-        coeff*new(cleanterms)
+        coeff*new(flattenop(*,cleanterms))
```

Comment:
I am not sure this is the best order of operations. You might have `(c*A) * (c*((c*A)*(c*A)))` -- for this particular example, flattening followed by prefactor extraction would work better. But I am not sure how general my statement is.

A second request: I know this state is kinda inherited already, but now that we are doing more and more fancy processing on construction, I want us to be a bit more disciplined about what goes in the constructor and what goes in the `*` operator. May we just remove the custom constructor and move all of this to `*`? Any reason that is a bad idea?

Reply:
I don't see any reason why this would be a bad idea. My original thought process is that using inner constructor methods would be more idiomatic, but perhaps you're right in that it is too fancy for our purposes. It would also make the source code more clear to future developers if we do the processing in the calls to `*`, `+`, etc.

Reply:
Thanks! Please proceed with that plan. For more context: one thing that is pestering the back of my mind is how fancy constructors plagued the development of SymPy. At some point, for various reasons related to more advanced simplification rules, it became useful to be able to construct the most raw unsimplified versions of expressions, but all the `__init__` methods were already fancy, so a lot of work had to be put in using another set of "pre-init" methods.

Reply:
Ah, that's interesting. Just out of curiosity, would there be any performance benefits in simplifying the constructors and moving the processing to conversion functions? I can't find any discussion about this in Julia discourse or on GitHub issue pages.

Reply:
I do not think there would be anything (besides potentially avoiding situation where we happen to write repetitive canonicalization steps). Nothing inherent to the language though (constructors and other functions are treated the same way by the compiler, which is also why we are permitted to return object of type A from the constructor of type B (strictly speaking we are not really constructing type B anymore))

However, this separation of concerns we are discussing above might in some situation make the functions more "type stable" which permits the compiler to make a bunch of optimizations (mostly removing runtime type checks and allocations).

Reply:
Thanks, this is good to know.